### PR TITLE
Respond with RAML instance type to killed pod instance.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
The RAML definition for the Pod instance killing defines a pod status
for the return type. This change will change the RAML type use that.

JIRA issues: MARATHON-8524

### CODE REVIEW

File: docs/docs/rest-api/public/api/v2/types/instance.raml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      attributes: offer.AgentAttribute[]
+  InstanceState:
+    type: object
+    properties:
+      condition: strings.Condition
+      since: datetime
+      activeSince?: datetime
+      healthy?: boolean
+      goal: strings.Goal
+  Instance:
```
Comment (jeschkies) [inline docs/docs/rest-api/public/api/v2/types/instance.raml]: The type is missing `tasksMap`, the instance id as an object and reservations. I do not think we should try too hard to be backwards compatible.

Comment (jeschkies) [reply docs/docs/rest-api/public/api/v2/types/instance.raml]: I think it is fine though. `reservations` is optional, the `tasksMap` can be empty. so the only breaking change is the `instanceId: { idString: "..." }`.

**Comment by jeschkies** [comment]
@timcharper, could you revisit the review?

**Comment by meln1k** [comment]
Isn't it breaking the backwards compatibility?

**Comment by meln1k** [comment]
Isn't it breaking our backwards compatibility?

**Comment by meln1k** [comment]
Isn't it breaking our backwards compatibility?

**Comment by jeschkies** [comment]
> Isn't it breaking our backwards compatibility?

Yes and now. As I mentioned in the comments we only break it for `instanceId: { idString: .. }`. Since `tasksMap` is a map it is valid to be empty.

**Comment by timcharper** [comment]
Okay, I've compared the output with 1.6.x we're missing `tasksMap`. Will it be too hard to keep that field, @jeschkies ? I understand it's unlikely that anyone will be depending on it, and, removing a JSON object is a breaking change.

**Comment by jeschkies** [comment]
> ... removing a JSON object is a breaking change.

IMHO the absence of `tasksMap` should be handled as an empty map. However, I can see that users would rely on it.

> Will it be too hard to keep that field, @jeschkies ?
I'll check. It's a little messy since it's another task format in addition to `EnrichedTask`. Our API is such a mess...

**Pull request merged**